### PR TITLE
Implement support for parameter refs

### PR DESCRIPTION
**Pull request by kpmg-jp-jnorton**
Implement support for parameter refs

### CODE REVIEW

File: core/src/martian/schema.cljc [hunk 1]
```diff
@@ -85,21 +85,26 @@
...
+               (make-schema definitions {} param)}))
        (into {})))
 
-(defn- resolve-ref [definitions ref]
-  (some->> ref
-           (re-find #"#/definitions/(.*)")
-           second
-           keyword
-           definitions))
+(defn- resolve-ref [definitions parameters ref]
```
Comment (oliyh) [inline core/src/martian/schema.cljc:91]: It feels like 'definitions' and 'parameters' are two categories of the same thing. I'd be tempted to merge them, either directly (risk of key collision?) or to provide it as a map that looks like `{:definitions {...} :parameters {...}}` and call it `ref-lookup`

File: core/src/martian/schema.cljc [hunk 2]
```diff
@@ -85,21 +85,26 @@
...
+  (if-let [d (re-find #"#/definitions/(.*)" ref)]
```
Comment (oliyh) [inline core/src/martian/schema.cljc:92]: Then on this line you could change the regex to be
`(let [[_ category k] (re-find #"#/(definitions|parameters)/(.*)" ref)]
  (get-in ref-lookup [(keyword category) (keyword k)]))`

I'm not sure if 'category' is the best word to describe it.

**Comment by oliyh** [comment]
Hi,

Thanks so much for this contribution. You are on the right lines, I just left some feedback that would help simplify it. Some unit tests would be great, and then I'd be happy to merge it.

Cheers
Oliy

File: core/src/martian/schema.cljc (hunk 1 above)
Comment (kpmg-jp-jnorton) [reply core/src/martian/schema.cljc]: Please review the most recent comment.  Thank you.

File: core/src/martian/schema.cljc (hunk 2 above)
Comment (kpmg-jp-jnorton) [reply core/src/martian/schema.cljc]: Please review the most recent comment.  Thank you.

File: core/src/martian/swagger.cljc [hunk 3]
```diff
@@ -47,23 +47,23 @@
...
 
 (defn- ->handler
-  [{:keys [definitions] :as swagger-map}
+  [{:keys [definitions parameters] :as swagger-map}
    path-item-parameters
    url-pattern
    [method swagger-definition]]
   (when-let [route-name (some-> (:operationId swagger-definition) ->kebab-case-keyword)]
-    (let [path-parts (tokenise-path url-pattern)
+    (let [ref-lookup {:definitions definitions :parameters parameters} path-parts (tokenise-path url-pattern)
```
Comment (oliyh) [inline core/src/martian/swagger.cljc:55]: This should be on its own line, and could be `(select-keys swagger-map [:definitions : parameters])`

**Comment by oliyh** [comment]
Hi,

This looks really good now, I just left one last minor comment.

Thanks

File: core/src/martian/swagger.cljc (hunk 3 above)
Comment (kpmg-jp-jnorton) [reply core/src/martian/swagger.cljc]: Completed.  Please review latest version.

**Pull request merged**